src/ar_storage.erl

```diff
@@ -49,52 +50,94 @@ ensure_directories() ->
 	filelib:ensure_dir(filename:join(DataDir, ?WALLET_LIST_DIR) ++ "/"),
 	filelib:ensure_dir(filename:join(DataDir, ?HASH_LIST_DIR) ++ "/").
 
-%% @doc Clear the cache of saved blocks.
-clear() ->
-	lists:map(fun file:delete/1, filelib:wildcard(filename:join([ar_meta_db:get(data_dir), ?BLOCK_DIR, "*.json"]))),
-	ar_block_index:clear().
+count_blocks_on_disk() ->
+	spawn(
+		fun() ->
+			DataDir = ar_meta_db:get(data_dir),
+			case file:list_dir(filename:join(DataDir, ?BLOCK_DIR)) of
+				{ok, List} ->
+					ar_meta_db:increase(blocks_on_disk, length(List));
+				{error, Reason} ->
+					ar:warn([
+						{event, failed_to_count_blocks_on_disk},
+						{reason, Reason}
+					])
+			end
+		end
+	).
 
-%% @doc Removes a saved block.
-delete_block(Hash) ->
-	ar_block_index:remove(Hash),
-	file:delete(block_filepath(Hash)).
+rename_block_files(HL) ->
+	BlocksDir = filename:join(ar_meta_db:get(data_dir), ?BLOCK_DIR),
+	case filelib:is_file(filename:join(BlocksDir, "blocks_renamed")) of
+		true ->
+			ok;
+		false ->
+			spawn(
+				fun() ->
+					ar:info([{event, start_rename_block_files}]),
+					lists:foldl(
+						fun(H, Height) ->
+							EncodedH = binary_to_list(ar_util:encode(H)),
+							LegacyName = filename:join(
+								BlocksDir,
+								integer_to_list(Height) ++ "_" ++ EncodedH ++ ".json"
+							),
+							NewName = filename:join(
+								BlocksDir,
+								EncodedH ++ ".json"
+							),
+							file:rename(LegacyName, NewName),
+							Height - 1
+						end,
+						length(HL) - 1,
+						HL
+					),
+					write_file_atomic(filename:join(BlocksDir, "blocks_renamed"), <<>>),
```

Comment:
Why do we need this `write_file_atomic` function?

Reply:
With atomic writes, the files do not get corrupted if the machine is halted in the process. Also, it allowes two nodes to safely share the storage.